lib/dao.js

```diff
@@ -3059,7 +3059,10 @@ DataAccessObject.replaceById = function(id, data, options, cb) {
 
         function replaceCallback(err, data) {
           if (err) return cb(err);
-
+          // if connector is cloudant, add the `_rev` property back before passing it to the user
+          if (connector.name === 'cloudant') {
+            context.data._rev = data._rev;
```

Comment:
It does not remove anything. If you scroll few lines down, it uses the **old** data in a sense it sends out the __**old rev**__ value back to the user upon replace. So we should update the `context.data._rev` point to the new `_rev` value  before sending it out.